File: src/World/Tile.java
package World;

import Utilities.Prefs;

public class Tile
{
	public int x, y, xIndex, yIndex, tileSize, maxFood, tileNumber, deadCooldown, cooldownThreshold,
		colorH, colorS, colorV;

	public double food, regenValue;
	
	boolean water;
	
	
	public Tile(int x, int y, int tileSize, int tileNumber, int xIndex, int yIndex, boolean water)
	{
		this.water = water;
		this.x = x;
		this.y = y;
		this.xIndex = xIndex;
		this.yIndex = yIndex;
		this.tileNumber = tileNumber;
		this.tileSize = tileSize;
		
		if(water) maxFood = 0;
		else maxFood = 100;
		if(!water) food = 60 + (int) (Math.random() * 40);
		if(water) colorH = 220;
		else colorH = (int)(food * 0.7 + 50);
		colorS = 80;
		colorV = 45;
		
		if(water) regenValue = 0;
		else regenValue = Prefs.TILE_REGEN_RATE;
		cooldownThreshold = Prefs.TILE_COOLDOWN_THRESH;
	}
	
	public void regenerateTileFood()
	{
		if(water) food = 0;
		if(water) return;
		
		colorH = (int) (food * 0.7 + 50);
		
		deadCooldown++;
		if(deadCooldown < cooldownThreshold) return;
		else deadCooldown = cooldownThreshold;
		
		if(food >= maxFood)
		{
			food = maxFood;
			return;
		}
		food += regenValue;
	}
	
	public void resetCooldown()
	{
		deadCooldown = 0;
	}
}

File: site/neural-net-app/tile.js
class Tile {
    constructor(xIndex, yIndex, tileId, water) {
        this.xIndex = xIndex;
        this.yIndex = yIndex;
        this.tileId = tileId;
        this.isWater = water;
        this.maxH = 120.0;

        this.x = this.xIndex * TILESIZE;
        this.y = this.yIndex * TILESIZE;

        if (this.isWater) {
            this.food = 0;
            this.colorH = 220;
        }
        else {
            this.food = randomBound(settings.tileMaxFood);
            this.colorH = Math.floor(70.0 * this.food / settings.tileMaxFood) + 50;
        }

        this.colorS = 80;
        this.colorV = 45;
        this.deadCooldown = settings.cooldownThreshold;
    }

    progress() {
        if (this.isWater) return;
        this.regenerateTileFood();
    }

    regenerateTileFood() {
        if (this.food >= settings.tileMaxFood) return;
        if (this.deadCooldown < settings.cooldownThreshold) {
            this.deadCooldown++;
            return;
        }
        this.food += settings.tile_regen_rate;
        this.colorH = Math.floor(70.0 * this.food / settings.tileMaxFood) + 50;
    }

    resetCooldown() {
        this.deadCooldown = 0;
    }
}
